wgpu-core/src/device/queue.rs

```diff
@@ -143,12 +143,56 @@ crate::impl_storage_item!(Queue);
 impl Drop for Queue {
     fn drop(&mut self) {
         resource_log!("Drop {}", self.error_ident());
+
+        let last_successful_submission_index = self
+            .device
+            .last_successful_submission_index
+            .load(Ordering::Acquire);
+
+        let fence = self.device.fence.read();
+        let wait_res = unsafe {
+            self.device.raw().wait(
+                fence.as_ref(),
+                last_successful_submission_index,
+                crate::device::CLEANUP_WAIT_MS,
+            )
+        };
+        drop(fence);
+
+        match wait_res {
+            Ok(true) => {}
+            // Note: If we don't panic here we are in UB land (destroying resources while they are still in use by the GPU).
+            Ok(false) => {
+                panic!("We timed out while waiting on the last successful submission to complete!");
+            }
+            Err(e) => {
+                panic!(
+                    "We ran into an error while waiting on the last successful submission to complete! - {e}"
+                );
+            }
```

Comment:
note: panicking in `Drop` can result in an abortion of the program if `Drop` was being called in the middle of a panic (see [`std::Drop`'s `Panics` section](https://doc.rust-lang.org/std/ops/trait.Drop.html#panics)).

Reply:
Actually I think aborting in this case is appropriate. If we don't, that's UB and we shouldn't continue unwinding.

Reply:
[Definitely not blocking](https://web.archive.org/web/20241002184501/https://conventionalcomments.org/#labels), by any means. Running into an abort here means (1) a user was panicking for some reason and (2) an error was returned in a function that has no means of dealing with an error.

If somebody runs into errors, do we have a clear workaround identified? If so, I'm a _lot_ less concerned about this, though I will not block even if there is not (since I know there's yet more follow-up work to do that should resolve it anyway).

Reply:
I implemented a retry mechanism for the cases that we can recover from. I can't think of other ways to deal with these errors.